### PR TITLE
Update base for Docker, remove python2

### PR DESCRIPTION
**Pull request by XVilka**
Last missing piece for [Python 2 removal](https://github.com/facebook/infer/issues/934) also updated Docker base for the latest Debian Stable - Buster. Also selected 4.08.1 for the Java build as suggested i https://github.com/facebook/infer/pull/1154#issuecomment-601144234

### CODE REVIEW

File: docker/master/Dockerfile [hunk 1]
```diff
@@ -63,9 +60,9 @@ RUN cd /infer && \
 
 FROM debian:stretch-slim AS executor
 
-# Install python 2.7 since infer requires it to run
+# Install python 3.7 since infer Clang plugins requires it to run
```
Comment (jvillard) [inline docker/master/Dockerfile:63]: To be pedantic, the plugins only need python for the build I think.

File: docker/master-java/Dockerfile [hunk 2]
```diff
@@ -18,19 +18,16 @@ RUN apt-get update && \
       libmpfr-dev \
       libsqlite3-dev \
       make \
-      openjdk-8-jdk-headless \
+      openjdk-11-jdk-headless \
```
Comment (jvillard) [inline docker/master-java/Dockerfile:21]: Did you test if infer works with java11? Eg `infer -- javac examples/Hello.java`.

**Comment by jvillard** [comment]
Looks good, thanks!

**Comment by XVilka** [comment]
There seems to be a bug with Debian/Ubuntu Python:
- https://bugs.launchpad.net/ubuntu/+source/apt-btrfs-snapshot/+bug/1763923
- https://github.com/nodejs/node/issues/30189
So it needs python3-distutils installation, also infer needs sqlite3.

File: docker/master-java/Dockerfile (hunk 2 above)
Comment (XVilka) [reply docker/master-java/Dockerfile:21]: Yes, it works.

File: docker/master/Dockerfile (hunk 1 above)
Comment (XVilka) [reply docker/master/Dockerfile]: Changed this line

File: docker/master/Dockerfile [hunk 3]
```diff
@@ -63,9 +61,7 @@ RUN cd /infer && \
 
 FROM debian:stretch-slim AS executor
```
Comment (skcho) [inline docker/master/Dockerfile:62]:
```suggestion
FROM debian:buster-slim AS executor
```

**Pull request closed without merging**